test/utilities.client.test.ts

```diff
@@ -1271,6 +1271,99 @@ describe('utilities.client', () => {
                 expect(errors[0].message).toEqual('The field schema.properties.field is a secret field, but its schema has changed. Please update the value in the input editor.');
             });
         });
+
+        describe('special cases for sub-schema', () => {
+            it('should allow sub-schema for object property', () => {
+                const { inputSchema, validator } = buildInputSchema({
+                    field: {
+                        title: 'Field title',
+                        description: 'My test field',
+                        type: 'object',
+                        editor: 'schemaBased',
+                        properties: {
+                            key1: {
+                                type: 'string',
+                                title: 'Key 1',
+                                description: 'Description for key 1',
+                                editor: 'textfield',
+                            },
+                            key2: {
+                                type: 'string',
+                                title: 'Key 2',
+                                description: 'Description for key 2',
+                                editor: 'textfield',
+                            },
+                        },
+                        additionalProperties: false,
+                        required: ['key1'],
+                    },
+                });
+                const inputs = [
+                    // 4 invalid inputs
+                    { field: [] },
+                    { field: {} },
+                    { field: { key2: 'value' } },
+                    { field: { key3: 'value' } },
+                    // 2 valid inputs
+                    { field: { key1: 'value' } },
+                    { field: { key1: 'value', key2: 'value' } },
+                ];
+
+                const results = inputs
+                    .map((input) => validateInputUsingValidator(validator, inputSchema, input))
+                    .filter((errors) => errors.length > 0);
+
+                // There should be 4 invalid inputs
+                expect(results.length).toEqual(4);
```

Comment:
Suggestion: Would change to explicit `validInputs` and `invalidInputs` (it's easier to read and makes the test condition stronger)